cmdx.py

```diff
@@ -2674,20 +2674,27 @@ def __str__(self):
         return str(self.read())
 
     def __repr__(self):
+        try:
+            # Delegate the value reading to __str__
+            read_result = str(self)
+            valid = True
+        except:
+            valid = False
+
         cls_name = '{}.{}'.format(__name__, self.__class__.__name__)
-        if self._mplug.attribute().apiType() == om.MFn.kCompoundAttribute:
-            return '{}("{}", "{}")'.format(cls_name,
-                                           self.node().name(),
-                                           self.name())
-        read_val = self.read()
-        if isinstance(read_val, string_types):
-            # Add surrounding single quotes, indicating the value is a string
-            read_val = '"{}"'.format(read_val)
-
-        return '{}("{}", "{}") == {}'.format(cls_name,
-                                             self.node().name(),
-                                             self.name(),
-                                             read_val)
+        msg = '{}("{}", "{}")'.format(cls_name,
+                                      self.node().name(),
+                                      self.name())
+        if valid:
+            try:
+                if self.typeClass() == String:
+                    # Add surrounding quotes, indicating it is a string
+                    read_result = '"{}"'.format(read_result)
+            except TypeError:
+                pass
+            msg += ' == {}'.format(read_result)
```

Comment:
Had to add this in order to avoid cases where it shows a valid value, but of an empty string:
```python
>>> myNode["myStr"]
cmdx.Plug("myNode", "myStr") == 
```